graphql/provider.go

```diff
@@ -40,10 +62,66 @@ func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{},
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
+
+	if oauth2LoginQueryValueAttribute := d.Get("oauth2_login_query_value_attribute"); d.Get("oauth2_login_query") != "" && oauth2LoginQueryValueAttribute != "" {
+		queryResponse, resBytes, err := queryExecute(ctx, d, config, "oauth2_login_query", "oauth2_login_query_variables")
+		if err != nil {
+			return nil, diag.FromErr(fmt.Errorf("unable to execute oauth2_login_query: %w", err))
+		}
+
+		if queryErrors := queryResponse.ProcessErrors(); queryErrors.HasError() {
+			return nil, *queryErrors
+		}
+
+		var queryResponseCty cty.Value
+		if queryResponseCty, err = gocty.ToCtyValue(string(resBytes), cty.String); err != nil {
+			return nil, diag.FromErr(err)
+		}
+
+		evalCtx := &hcl.EvalContext{
+			Variables: map[string]cty.Value{
+				"oauth2_login_query_response": queryResponseCty,
+			},
+			Functions: map[string]function.Function{
+				"jsondecode": stdlib.JSONDecodeFunc,
+			},
+		}
+
+		var expr hcl.Expression
+		var hclDiags hcl.Diagnostics
+		valueTemplate := fmt.Sprintf("${jsondecode(oauth2_login_query_response).%s}", oauth2LoginQueryValueAttribute.(string))
```

Comment:
I like this approach for this particular use-case of getting the attribute's value, but I am debating with myself whether or not I want to use the same code used by other resp key/value parsers in this project. In keys.go, we have 
`buildResourceKeyArgs` and `getResourceKey`

Perhaps we can use a helper function in this file that uses that same functionality which will behave the same way the `compute_mutation_keys` behaves in resources.

```go
func getOAuth2LoginQueryAttributeValue(attribute string, data map[string]interface) string {
		resourceKeyArgs := buildResourceKeyArgs(attribute)
		value, err := getResourceKey(data, resourceKeyArgs...)
		if err != nil {
			return nil, err
		}
		 return value.(string)
}
```
Something like that should work, and keep things consistent throughout the project.

Reply:
Just to add to this, the hcl parser works great for this use-case, no doubt. I'm primarily concerned about using the same approach everywhere in the project so that if something breaks with one or the other, Im not confused. The functions in `keys.go` are specialized for parsing a key's value in a recursive N level deep object, so most likely overkill to get an access token from an OAuth response, but yeah.. just keeps functionality and behavior the same everywhere.

Reply:
I wasn't aware of the existence of these functions, and have now updated the code to use them.

Reply:
No worries. Thanks for the update